README.md

```diff
@@ -32,6 +32,21 @@ public class ExampleApplication extends Application {
 
 **You're good to go!** LeakCanary will automatically show a notification when an activity memory leak is detected in your debug build.
 
+## Releases
```

Comment:
Why move this into README? Adds more content to an already long readme.

I'm especially worried about the snapshot part. It doesn't mention the current snapshot version so that's extra work to figure out. The "snapshot" section is gone in the CHANGELOG. The point of that section was that we could build it progressively by adding a line there with every important PR. Granted we didn't do a good job of that for this beta, but before that we had it.